File: app/src/main/java/com/robotnec/budget/app/adapters/support/BaseAdapter.java
package com.robotnec.budget.app.adapters.support;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;

import java.util.ArrayList;
import java.util.List;

/**
 * @author zak <zak@swingpulse.com>
 */
public abstract class BaseAdapter<T, S extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<S> {

    protected final Context context;
    protected final List<T> items;
    protected final LayoutInflater layoutInflater;

    public BaseAdapter(Context context) {
        items = new ArrayList<>();
        this.context = context;
        layoutInflater = LayoutInflater.from(context);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public void update(List<T> items) {
        this.items.clear();
        this.items.addAll(items);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/robotnec/budget/app/adapters/support/BaseAdapter.kt
package com.robotnec.budget.app.adapters.support

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater

import java.util.ArrayList

/**
 * @author zak zak@swingpulse.com>
 */
abstract class BaseAdapter<T, S : RecyclerView.ViewHolder>(
        protected val context: Context) : RecyclerView.Adapter<S>() {
    protected val items: MutableList<T>
    protected val layoutInflater: LayoutInflater

    init {
        items = ArrayList<T>()
        layoutInflater = LayoutInflater.from(context)
    }

    override fun getItemCount(): Int {
        return items.size
    }

    fun update(items: List<T>) {
        this.items.clear()
        this.items.addAll(items)
        notifyDataSetChanged()
    }
}
